solidity/test/TokenHolder.js

```diff
@@ -1,56 +1,169 @@
-    });
+                    const balance = await getBalance(tokenAddress, receiver);
+                    expect(balance).to.be.bignumber.equal(prevBalance.add(amount));
+                });
 
-    it('should revert when attempting to withdraw tokens to an invalid account address', async () => {
-        await expectRevert(holder.withdrawTokens(erc20Token.address, ZERO_ADDRESS, new BN(1)), 'ERR_INVALID_ADDRESS');
-    });
+                it('should revert when a non-owner attempts to withdraw', async () => {
+                    await expectRevert(
+                        holder.withdrawTokens(tokenAddress, receiver, new BN(1), { from: nonOwner }),
+                        'ERR_ACCESS_DENIED'
+                    );
+                });
+
+                it('should revert when attempting to withdraw from an invalid asset address', async () => {
+                    await expectRevert(
+                        holder.withdrawTokens(ZERO_ADDRESS, receiver, new BN(1)),
+                        'Address: call to non-contract'
+                    );
+                });
 
-    it('should revert when attempting to withdraw tokens to the holder address', async () => {
-        await expectRevert(holder.withdrawTokens(erc20Token.address, holder.address, new BN(1)), 'ERR_ADDRESS_IS_SELF');
+                it('should revert when attempting to withdraw tokens to an invalid account address', async () => {
+                    await expectRevert(
+                        holder.withdrawTokens(tokenAddress, ZERO_ADDRESS, new BN(1)),
+                        'ERR_INVALID_ADDRESS'
+                    );
+                });
+
+                it('should revert when attempting to withdraw an amount greater than the holder balance', async () => {
+                    const balance = await getBalance(tokenAddress, holder.address);
+                    const amount = balance.add(new BN(1));
+
+                    if (isETH) {
+                        await expectRevert.unspecified(holder.withdrawTokens(tokenAddress, receiver, amount));
+                    } else {
+                        await expectRevert(
+                            holder.withdrawTokens(tokenAddress, receiver, amount),
+                            'SafeMath: subtraction overflow'
+                        );
+                    }
+                });
+            });
+        }
     });
 
-    it('should revert when attempting to withdraw an amount greater than the holder balance', async () => {
-        const balance = await erc20Token.balanceOf.call(holder.address);
+    describe('withdraw multiple assets', () => {
+        let tokenAddresses;
+        let amounts;
+
+        beforeEach(async () => {
+            tokenAddresses = [NATIVE_TOKEN_ADDRESS, token.address, token2.address];
+            amounts = {};
+
+            for (const tokenAddress of tokenAddresses) {
+                amounts[tokenAddress] = new BN(100);
```

Comment:
we should also test transferring different amounts (amount * index?)

Reply:
👍